src/sentry/tasks/summaries/weekly_reports.py

```diff
@@ -262,20 +262,23 @@ def prepare_organization_report(
 @dataclass(frozen=True)
 class OrganizationReportBatch:
     ctx: OrganizationReportContext
-    batch_id: uuid.UUID
+    batch_id: uuid.UUID | str
 
     dry_run: bool = False
-    target_user: User | None = None
+    target_user: User | int | None = None
```

Comment:
Because this isn't being sent into the task parameters, you could remove `User` from the type and always use `target_user_id`.

Reply:
Why is this the case? If our new deployment of this celery worker (which accepts both `User` and `int`) picks up an old task, wouldn't `OrganizationReportBatch` still need to accept `User` still?